src/iop/temperature.c

```diff
@@ -938,10 +926,17 @@ void reload_defaults(dt_iop_module_t *module)
   // we might be called from presets update infrastructure => there is no image
   if(!module->dev) goto end;
 
+  const int is_raw = dt_image_is_raw(&module->dev->image_storage);
+
+  // White balance module doesn't need to be enabled for monochrome raws (like
+  // for leica monochrom cameras). prepare_matrices is a noop as well, as there
+  // isn't a color matrix, so we can skip that as well.
+  if(is_raw && dt_image_is_monochrome(&(module->dev->image_storage))) goto gui;
+
   if(module->gui_data) prepare_matrices(module);
 
   /* check if file is raw / hdr */
-  if(dt_image_is_raw(&module->dev->image_storage))
+  if(is_raw)
   {
     // raw images need wb:
     module->default_enabled = 1;
```

Comment:
I'm wondering if the "enable" button should be disabled for these monochrome raws, especially since it can't work at all for them.

Reply:
I personally think it should be disabled.

Where are the default module orders defined, so I can go through them? Is it `develop/format.c`?

Reply:
Right sidepanel, from bottom to top :)

Reply:
Oh, found it, it's in `iop_dependencies.py`

Reply:
BTW, by hidden i mean https://github.com/kelvie/darktable/blob/a05f425020e03ab1d9e6942d86775ec3df23f2b8/src/iop/demosaic.c#L4659